ddtrace/contrib/urllib3/patch.py

```diff
@@ -0,0 +1,166 @@
+
+    :param args: Positional arguments
+    :param kwargs: Keyword arguments
+    :param pos: The positional index of the argument if passed in as a positional arg
+    :param kw: The name of the keyword if passed in as a keyword argument
+    :param default: The default value to return if the argument was not found in args or kwaergs
+    :return: The value of the target argument
+    """
+    if kw in kwargs:
+        return kwargs[kw]
+
+    if pos < len(args):
+        return args[pos]
+
+    return default
+
+
+def _wrap_urlopen(func, obj, args, kwargs):
+    """
+    Wrapper function for the lower-level urlopen in urllib3
+
+    :param func: The original target function "urlopen"
+    :param obj: The patched instance of ``HTTPConnectionPool``
+    :param args: Positional arguments from the target function
+    :param kwargs: Keyword arguments from the target function
+    :return: The ``HTTPResponse`` from the target function
+    """
+    request_method = _infer_argument_value(args, kwargs, 0, "method")
+    request_url = _infer_argument_value(args, kwargs, 1, "url")
+    request_headers = _infer_argument_value(args, kwargs, 3, "headers")
+    request_retries = _infer_argument_value(args, kwargs, 4, "retries")
+
+    # HTTPConnectionPool allows relative path requests; convert the request_url to an absolute url
+    if request_url.startswith("/"):
+        request_url = parse.urlunparse(
+            (
+                obj.scheme,
+                "{}:{}".format(obj.host, obj.port) if obj.port and obj.port not in DROP_PORTS else str(obj.host),
+                request_url,
+                None,
+                None,
+                None,
+            )
+        )
+
+    parsed_uri = parse.urlparse(request_url)
+    hostname = parsed_uri.netloc
+    sanitized_url = sanitize_url_for_tag(request_url)
+
+    tracer = getattr(obj, "datadog_tracer", ddtrace.tracer)
+
+    if not tracer.enabled:
+        return func(*args, **kwargs)
+
+    with tracer.trace("urllib3.request", span_type=SpanTypes.HTTP) as span:
+
+        span.service = _extract_service_name(span, hostname, config.urllib3["split_by_domain"])
+
+        # If distributed tracing is enabled, propagate the tracing headers to downstream services
+        if config.urllib3["distributed_tracing"]:
+            if request_headers is None:
+                request_headers = {}
+                kwargs["headers"] = request_headers
+            propagator = HTTPPropagator()
+            propagator.inject(span.context, request_headers)
+
+        store_request_headers(request_headers, span, config.urllib3)
+        span.set_tag(http.METHOD, request_method)
+        span.set_tag(http.URL, sanitized_url)
+        if config.urllib3["trace_query_string"]:
+            span.set_tag(http.QUERY_STRING, parsed_uri.query)
+        if config.urllib3["analytics_enabled"]:
+            span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.urllib3["analytics_sample_rate"])
```

Comment:
We can use the `IntegrationConfig.get_analytics_sample_rate()` function here:

```suggestion
span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.urllib3.get_analytics_sample_rate())
```